### ISSUE TITLE
Errors with ZFS on pre and post restart script

### ISSUE THREAD

**Issue by elalann** [nextcloud/vm#1766]
Hello there,

I was trying the script on a Ubuntu 20.04.1 server, and I saw "zpool export ncdata failed" but the script continued his work without any problem.
The ONLY thing I did before trying the script it's following this guide:
https://ubuntu.com/tutorials/setup-zfs-storage-pool
because I didn't know the script made the ZFS pool, but after I noticed that, I did destroy the pool I created eralier and then

When I restarted this happened:

```
+-----------------------------------------------------------------------+
[sudo] password for admints:
Fetching all the variables from lib.sh...
  Logical volume "NcVM-installation" successfully removed
  Logical volume "NcVM-startup" created.
Testing if network is OK...
Waiting for network to restart...
Checking connection...
Online!
Testing if network is OK...
Waiting for network to restart...
Checking connection...
Online!
cannot import 'ncdata': no such pool available
  sdb       99.5G   148K  99.5G        -         -     0%  0.00%      -  ONLINE
/dev/sda: gpt partitions 1 2 3
/dev/sdb: gpt partitions 1 9
/dev/mapper/ubuntu--vg-NcVM--startup: loop partitions 1
/dev/mapper/ubuntu--vg-ubuntu--lv-real: loop partitions 1
/dev/mapper/ubuntu--vg-NcVM--startup-cow: msdos partitions
/dev/mapper/ubuntu--vg-ubuntu--lv: loop partitions 1
cannot open 'ncdata': no such pool
Sorry but something went wrong. Please report this issue to https://github.com/nextcloud/vm/issues and include the output of the error message. Thank you!
zpool export ncdata failed
Posting notification to users that are admins, this might take a while...
Posting 'Sorry but something went wrong. Please report this issue to https://github.com/nextcloud/vm/issues and include the output of the error message. Thank you!' to ncadmin
root@nextcloud:~#
```

I will reinstall the system and run the script with the system 100% clean, I'm making this issue only to help make the script better, if it doesn't, feel free to close it without any explanation :)

Regards,
Alan.

**Comment by szaimen** [comment]
cc @enoch85

**Comment by enoch85** [comment]
@elalann You shouldn't touch anything in the VM (well, if you're not an expert Linux hacker) Everything is setup as it should already.

Please umount the VM, destroy it, then import it again and start over.

**Comment by enoch85** [comment]
@elalann Most likely case, you created a ZFS pool which a different name than "ncdata". Can you please confirm?

**Comment by elalann** [comment]
Hello guys, @szaimen and @enoch85.
Indeed I created a ZFS pool, with the name "ncdata" but I destroyed it before re-running the script.

Not really sure if the zfs tool didn't destroy it properly, or what actually happened, because as soon the script finished the errors came with it and it was simple as "right click > delete vm" and start over again.

This actually happened because I didn't knew the script itself did the job of creating the ncdata zfs, so as soon I saw an error message about "ncdata pool exists" or something similar, I stopped the script, destroyed the pool and started back again after a reboot.

Regards,
Alan.
